test/src/test/java/jenkins/model/JenkinsBuildsAndWorkspacesDirectoriesTest.java

```diff
@@ -0,0 +1,364 @@
+            final File configFile = new File(rule.jenkins.getRootDir(), "config.xml");
+            final String screwedUp = FileUtils.readFileToString(configFile).
+                    replaceFirst("<buildsDir>.*</buildsDir>", "<buildsDir>eeeeeeeeek</buildsDir>");
+            FileUtils.write(configFile, screwedUp);
+        });
+
+        story.thenDoesNotStart();
+    }
+
+    @Issue("JENKINS-50164")
+    @Test
+    public void buildsDir() throws Exception {
+        loggerRule.record(Jenkins.class, Level.WARNING)
+                .record(Jenkins.class, Level.INFO)
+                .capture(100);
+
+        story.then(step -> {
+                       assertFalse(logWasFound("Using non default builds directories"));
+                   }
+        );
+
+        story.then(steps -> {
+            assertTrue(story.j.getInstance().isDefaultBuildDir());
+            setBuildsDirProperty("$JENKINS_HOME/plouf/$ITEM_FULL_NAME/bluh");
+            assertFalse(JenkinsBuildsAndWorkspacesDirectoriesTest.this.logWasFound(LOG_WHEN_CHANGING_BUILDS_DIR));
+        });
+
+        story.then(step -> {
+                       assertFalse(story.j.getInstance().isDefaultBuildDir());
+                       assertEquals("$JENKINS_HOME/plouf/$ITEM_FULL_NAME/bluh", story.j.getInstance().getRawBuildsDir());
+                       assertTrue(logWasFound("Changing builds directories from "));
+                   }
+        );
+
+        story.then(step -> assertTrue(logWasFound("Using non default builds directories"))
+        );
+    }
+
+    @Issue("JENKINS-50164")
+    @Test
+    public void workspacesDir() throws Exception {
+        loggerRule.record(Jenkins.class, Level.WARNING)
+                .record(Jenkins.class, Level.INFO)
+                .capture(1000);
+
+        story.then(step -> assertFalse(logWasFound("Using non default workspaces directories")));
+
+        story.then(step -> {
+            assertTrue(story.j.getInstance().isDefaultWorkspaceDir());
+            final String workspacesDir = "bluh";
+            setWorkspacesDirProperty(workspacesDir);
+            assertFalse(logWasFound("Changing workspaces directories from "));
+        });
+
+        story.then(step -> {
+            assertFalse(story.j.getInstance().isDefaultWorkspaceDir());
+            assertEquals("bluh", story.j.getInstance().getRawWorkspaceDir());
+            assertTrue(logWasFound("Changing workspaces directories from "));
+        });
+
+
+        story.then(step -> {
+                       assertFalse(story.j.getInstance().isDefaultWorkspaceDir());
+                       assertTrue(logWasFound("Using non default workspaces directories"));
+                   }
+        );
+    }
+
+    @Issue("JENKINS-50164")
+    @LocalData
+    @Test
+    public void fromPreviousCustomSetup() {
+
+        assumeFalse(Functions.isWindows()); // Default Windows lifecycle does not support restart.
```

Comment:
NIT: it is much nicer to use `assumeFalse(" Default Windows lifecycle does not support restart.", Function.isWindows())` as then the reason for the skipped test will be reported.

Reply:
Fixed